kong/plugins/ldap-auth/access.lua

```diff
@@ -84,14 +86,31 @@ local function load_credential(given_username, given_password, conf)
   return {username = given_username, password = given_password}
 end
 
+
+local function acl_cache_key(conf, username)
+  local cache_key = {
+    "ldap_auth_cache",
+    md5(concat({
+      conf.ldap_host,
+      conf.ldap_port,
+      conf.base_dn,
+      conf.attribute,
+      username,
+      conf.cache_ttl,
+    }, ":"))
+  }
+
+  return concat(cache_key, ":")
```

Comment:
It is a lot more efficient to follow the approach taken by `dao:cache_key()` rather than paying the cost of 2 shirt-lived tables, the double concatenation operation, and the `md5` hash. Any reason for now using a simple `string.format` instead? Combined with @Tieske's comment about caching the cache key prefix (that of the plugin configuration), and simply use the `username` as the dynamic component?

Reply:
Sure, I will change that. A lot more in this case might be a few microseconds, but of course better to do it like that. Not sure though, should we use something else than `MD5`. Maybe requiring `lua-resty-string` and using `SHA256` instead.

Reply:
Well, as always, it isn't about making that particular piece of the code run a few microseconds faster, but rather prevent the spreading of anti-patterns and ensure that _overall_, the proxy still performs relatively well in hot code paths. To avoid accumulating performance penalties when they can be avoided. It is a continuous effort.
In this case if we cache the key and only rotate the `username` part, we can get away with the current approach, sure.

I'm not sure why the hashing is required though?

Reply:
I think the resulting string is a lot smaller?

Reply:
That's the only benefit I am seeing as well (in which case md5 should be just fine?)

Reply:
A key without hashing:
```
ldap_auth_cache:ec2-54-172-82-117.compute-1.amazonaws.com:389:ou=scientists,dc=ldap,dc=mashape,dc=com:uid:2:einstein
```

A key with hashing:
```
ldap_auth_cache:aec022a3d1d6590c4432b4bc65433e2e:einstein
```